gmso/formats/lammpsdata.py

```diff
@@ -225,3 +235,265 @@ def write_lammpsdata(topology, filename, atom_style='full'):
+    atom_lines = open(filename, 'r').readlines()[i+2:i+n_atoms+2]
+    for line in atom_lines:
+        atom = line.split()
+        atom_type = atom[2]
+        charge = u.unyt_quantity(float(atom[3]), get_units(unit_style)['charge'])
+        coord = u.angstrom * u.unyt_array([
+            float(atom[4]),
+            float(atom[5]),
+            float(atom[6])])
+        site = Site(
+            charge=charge,
+            position=coord,
+            atom_type=type_list[int(atom_type)-1]
+            )
+        element = element_by_mass(site.atom_type.mass.value)
+        site.name = element.name
+        site.element = element
+        topology.add_site(site)
+
+    topology.update_sites()
+
+    return topology
+
+def _get_box_coordinates(filename, unit_style, topology):
+    """Function to parse box information
+    """
+    with open(filename, 'r') as lammps_file:
+        for line in lammps_file:
+            if 'xlo' in line.split():
+                break
+        x_line = line.split()
+        y_line = lammps_file.readline().split()
+        z_line = lammps_file.readline().split()
+
+        x =  float(x_line[1])-float(x_line[0])
+        y =  float(y_line[1])-float(y_line[0])
+        z =  float(z_line[1])-float(z_line[0])
+
+        # Check if box is triclinic
+        tilts = lammps_file.readline().split()
+        if 'xy' in tilts:
+            xy = float(tilts[0])
+            xz = float(tilts[1])
+            yz = float(tilts[2])
+           
+            xhi = float(x_line[1]) - np.max([0.0, xy, xz, xy+xz])
+            xlo = float(x_line[0]) - np.min([0.0, xy, xz, xy+xz])
+            yhi = float(y_line[1]) - np.max([0.0, yz])
+            ylo = float(y_line[0]) - np.min([0.0, yz])
+            zhi = float(z_line[1])
+            zlo = float(z_line[0])
+
+            lx = xhi - xlo
+            ly = yhi - ylo
+            lz = zhi - zlo
+
+            c = np.sqrt(lz**2 + xz**2 + yz**2)
+            b = np.sqrt(ly**2 + xy**2)
+            a = lx
+
+            alpha = np.arccos((yz*ly+xy*xz)/(b*c))
+            beta = np.arccos(xz/c)
+            gamma = np.arccos(xy/b)
+
+            # Box Information
+            lengths = u.unyt_array([a, b, c], get_units(unit_style)['distance'])
+            angles = u.unyt_array([alpha, beta, gamma], u.radian)
+            angles.to(get_units(unit_style)['angle'])
+            topology.box=Box(lengths, angles)
+        else:
+            # Box Information
+            lengths = u.unyt_array([x,y,z], get_units(unit_style)['distance'])
+            topology.box = Box(lengths)
+
+        return topology
+
+def _get_ff_information(filename, unit_style, topology):
+    """Function to parse atom-type information
+    """
+    with open(filename, 'r') as lammps_file:
+        types = False
+        for i, line in enumerate(lammps_file):
+            if 'atom' in line:
+                n_atomtypes = int(line.split()[0])
+                types = True
+            elif 'Masses' in line:
```

Comment:
Kinda worried about capitalization in cases like these; I don't know how LAMMPS treats them (`1 atom types` vs `1 Atom types`). The LAMMPS documentation is a beast to navigate but it seems like it's lowercase in the header and capitalized (`Atoms`) later on when actually enumerating the data. So this should work

https://lammps.sandia.gov/doc/read_data.html

Reply:
I agree this is something we should keep an eye out for in the future.